src/enzo/ActiveParticleRoutines.C

```diff
@@ -0,0 +1,508 @@
+/***********************************************************************
+/
+/  ROUTINES FOR THE STAR PARTICLE CLASS
+/
+/  written by: John Wise
+/  date:       March, 2009
+/  modified1:  November, 2011 (converting into active particles)
+/
+/  PURPOSE: Instead of restricting star particles to the typical 
+/           particle attributes, this class gives more functionality 
+/           to them.
+/
+************************************************************************/
+#include "preincludes.h"
+#include "performance.h"
+#include "ErrorExceptions.h"
+#include "macros_and_parameters.h"
+#include "typedefs.h"
+#include "global_data.h"
+#include "Fluxes.h"
+#include "GridList.h"
+#include "ExternalBoundary.h"
+#include "Grid.h"
+#include "Hierarchy.h"
+#include "TopGridData.h"
+#include "LevelHierarchy.h"
+#include "ActiveParticle.h"
+#include "ActiveParticle_SmartStar.h"
+#include "phys_constants.h"
+#include "CosmologyParameters.h"
+#define MAX_TEMPERATURE 1e8
+#define FACTORINCREASE 1e10
+#define MAX_ENERGY 1e7
+#define DENSITY_WEIGHTED 1
+#define RAMPTIME 1e4
+/* We need to make sure that we can operate on the grid, so this dance is
+ * necessary to make sure that grid is 'friend' to this particle type. */
+
+class SmartStarGrid : private grid {
+  friend class ActiveParticleType_SmartStar;
+};
```

Comment:
Add comment about the purpose of this class.

Reply:
Clarified.